stand/libsa/strcasecmp.c

```diff
@@ -41,29 +41,37 @@ static char sccsid[] = "@(#)strcasecmp.c	8.1 (Berkeley) 6/4/93";
 int
 strcasecmp(const char *s1, const char *s2)
 {
-	const u_char
-			*us1 = (const u_char *)s1,
-			*us2 = (const u_char *)s2;
+	const u_char *us1 = (const u_char *)s1,
+	    *us2 = (const u_char *)s2;
 
-	while (tolower(*us1) == tolower(*us2++))
-		if (*us1++ == '\0')
+	u_char u1, u2;
+
+	while ((u1 = tolower(*us1)) == (u2 = tolower(*us2))) {
+		if (u1 == '\0') {
 			return (0);
-	return (tolower(*us1) - tolower(*--us2));
+		}
+		us1++, us2++;
+	}
+	return (u1 - u2);
```

Comment:
These are more extensive changes. It's likely not wrong, but should be separate since it's not a compiler workaround